src/observable/ScalarObservable.ts

```diff
@@ -80,8 +80,8 @@ proto.reduce = function <T, R>(project: (acc: R, x: T) => R, seed?: R): Observab
     return <any>this;
   }
   let result = tryCatch(project)(seed, this.value);
-  if (result === errorObject) {
-    return new ErrorObservable(errorObject.e);
+  if (result as any === errorObject) {
```

Comment:
first, my question was simply if there reason to choose between below :)

`result as any` vs `<any>result`

for typeof errorObject, not sure at this moment..since it's literally kind of  `any` object, so it could be typeof `any` explicitly though.